tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+- name: Create Synapse user
+  user:
+    name: "{{ synapse_user }}"
+    state: present
+
+- name: Prepare Synapse directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ synapse_user }}"
+  loop:
+    - "{{ synapse_data_dir }}"
+
+- name: Install Synapse dependencies
+  apt:
+    name:
+      - python-virtualenv
+  notify:
+    - Restart Synapse service
+
+- name: Install Synapse's python dependencies
+  become: true
+  become_user: "{{ synapse_user }}"
+  pip:
+    state: latest
+    virtualenv: "{{ synapse_install_dir }}/env"
+    name:
+      - pip
+      - setuptools
+      - matrix-synapse
+  notify:
+    - Restart Synapse service
+
+- name: Install Synapse REST Password provider
+  get_url:
+    url: https://raw.githubusercontent.com/kamax-matrix/matrix-synapse-rest-auth/master/rest_auth_provider.py
+    dest: "{{ synapse_install_dir }}/env/lib/python3.5/site-packages/rest_auth_provider.py"
```

Comment:
Please don't hardcode the python version

Reply:
We could read the site-packages from the virtualenv via `{{ synapse_install_dir }}/env/bin/python -m site`, but the output still has to be parsed.

But how does this work together with upgrading the python version? As it turns out, the ansible pip module (with virtualenv parameter) will not upgrade the python version of an existing virtualenv. For a freshly created virtualenv, by the way, it uses (by default) the python version which ansible has been run with. This is in our setup the system-wide installed python3 version.

Of course I think that whether or not the virtualenv is freshly created shouldn't make a difference, for the sake of purity. Therefore we first need to find a solution how to upgrade the virtualenv, as soon as a new python3 version is deployed system-wide. This solution should of course not encompass deleting and recreating the virtualenv on every deployment.

I think that we should solve that problem first. Its solution might change the possibilities how the problem with the hardcoded python version can be solved.

Reply:
Or we could just omit the virtualenv...

Reply:
This is now resolved. See my newest commit on this branch: 1a59cf84dcfaf84935d23f9d8bdc8106731373f0

Reply:
I cannot guarantee that the downloaded `matrix-synapse-rest-auth` plugin will work with other Python versions, though. Do you approve that this is resolved, nonetheless?